SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h

```diff
@@ -2,11 +2,49 @@
 
 #pragma once
 
+#include "Utils/SpatialBasicAwaiter.h"
 #include <WorkerSDK/improbable/c_worker.h>
+
 #include "SpatialWorkerFlags.generated.h"
 
-DECLARE_DYNAMIC_DELEGATE_TwoParams(FOnWorkerFlagsUpdatedBP, const FString&, FlagName, const FString&, FlagValue);
-DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnWorkerFlagsUpdated, const FString&, FlagName, const FString&, FlagValue);
+DECLARE_DYNAMIC_DELEGATE_TwoParams(FOnAnyWorkerFlagUpdatedBP, const FString&, FlagName, const FString&, FlagValue);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnAnyWorkerFlagUpdated, const FString&, FlagName, const FString&, FlagValue);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FOnWorkerFlagUpdatedBP, const FString&, FlagValue);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnWorkerFlagUpdated, const FString&, FlagValue);
+
+UENUM()
+enum class ESpatialCallbackStyle : uint8
```

Comment:
Elsewhere we had separate functions rather than an enum as it's not something you determine dynamically. If you think this is a lot cleaner that's fine although it means adding in more global space types.

Reply:
I can see your point, I went for the reducing the surface area / making it a bit cleaner approach.

Reply:
@samiwh split this back out as suggested